components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/vault/VaultTool.java

```diff
@@ -234,6 +234,8 @@ private static void addSystemProperties() {
         }
 
         String secureVaultRegex = properties.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
-        System.setProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY, secureVaultRegex);
+        if (secureVaultRegex.isEmpty()) {
+            System.setProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY, secureVaultRegex);
+        }
     }
```

Comment:
Could 'secureVaultRegex' ever be null?

Reply:
Yes. What happens is it gets the value from the carbon.property file first. 
If the property doesnt exist in the file it ends up as null and System.setProperty will throw a null pointer exception when it tries to set the property. 


With the null check is done, then it skips setting the property as null. And it uses the default regex in the previous function

Reply:
The previous function mentioned is here: https://github.com/wso2/micro-integrator/pull/650/files#diff-d9ed61ef411ddd14fd475144f74353ecR132 

This sets the proper regex if it cant find the property in the property file. 
```
String passwordRegex = System.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
if (Objects.isNull(passwordRegex)) {  
     System.out.println(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY + " property cannot be found in the carbon.properties file. Hence setting the default value as " + Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY + "=^[\\\\S]{5,30}$\n");
}
```

Reply:
Done. I removed the check for empty. In hindsight its not really required. My mistake. 
It is now just checking for null. 

The only requirement for isEmpty would be if the property is set without a regex value. It will simply state that the password does not match the regular expression.